src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -404,14 +406,21 @@ public void GetGrainTypeInfo(int typeCode, out string grainClass, out PlacementS
                     ActivationCollector.TryRescheduleCollection(result);
                     return result;
                 }
-
+                
                 int typeCode = address.Grain.GetTypeCode();
                 string actualGrainType = null;
+                MultiClusterRegistrationStrategy activationStrategy;
 
-                if (typeCode != 0) // special case for Membership grain.
-                    GetGrainTypeInfo(typeCode, out actualGrainType, out placement);
+                if (typeCode != 0)
+                {
+                    GetGrainTypeInfo(typeCode, out actualGrainType, out placement, out activationStrategy);
+                }
                 else
+                {
+                    // special case for Membership grain.
                     placement = SystemPlacement.Singleton;
+                    activationStrategy = MultiClusterRegistrationStrategy.GetDefault();
```

Comment:
I think it must be explicitly `ClusterLocalRegistration` for membership grain, independent of the default.